### PR TITLE
Feature/validators

### PR DESCRIPTION
**Pull request by j-mai**
This PR:
- Adds time and event validators in preparation for adding time elapsed parsers through configuration.
- Re-organizes files into different packages.

### CODE REVIEW

**Review by kcajmagic — APPROVED** [review]

File: event/parsing/eventValidator.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+			return false, fmt.Errorf("%w: %v", ErrInvalidBirthdate, err)
+		}
+		return false, fmt.Errorf("%w: %v", ErrInvalidBootTime, err)
+	}
+
+	return true, nil
+}
+
+// IsWRPValid checks if a wrp is valid baased on the rules kept by the EventValidator.
+func (e eventValidator) IsWRPValid(wrpWithTime queue.WrpWithTime) (bool, error) {
```
Comment (kristinaspring) [inline event/parsing/eventValidator.go:91]: I'm starting to think `eventValidator` doesn't really describe what this struct is doing? A few options:
- split up the validator into an `eventValidator` and `wrpValidator`
- somehow add `.Validate()` functions to the actual structs (haven't figured this one out yet - especially since we have configurable limits and shared functions betweeen them)
- drop the `event` from the name (really seems like the easiest option here)

The only good reason I can think of to split this into two different structs (that will have a lot of the same dependencies for time reasons) is if a consumer might only need to validate events or only need to validate wrps.

Comment (kristinaspring) [reply event/parsing/eventValidator.go:91]: Also - you could shorten the function name for brevity's sake to `ValidWRP` (and similarly `ValidEvent`).  I'm more ambivalent about this.

**Review by kristinaspring — CHANGES_REQUESTED** [review]
This was a little hard to follow, so if a comment doesn't make sense, it's probably because I missed something. 🙂 Let me know if a suggestion doesn't make sense or doesn't seem like a good idea.

File: event/parsing/eventValidator.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+var (
+	ErrInvalidEventType = errors.New("event type doesn't match")
+	ErrInvalidRegex     = errors.New("invalid regex")
+	ErrInvalidBootTime  = errors.New("invalid boot-time")
+	ErrInvalidBirthdate = errors.New("invalid birthdate")
+	ErrTimeParse        = errors.New("parsing error")
+)
+
+// NewEventValidation creates a new EventValidator from an EventRule
+func NewEventValidation(rule EventRule, validTo time.Duration, currentTime func() time.Time) (EventValidation, error) {
```
[the remark this answers is not in the document]
Comment (j-mai) [reply event/parsing/eventValidator.go:46]: I think one of the go tools suggested that returning an unexported struct might be annoying to use which was why I put the interface in this file and had the constructor return the interface instead of *eventValidator. But since the other package should only use eventValidator as the interface type, moving it there sounds good!

File: event/parsing/eventValidator.go (hunk 1 above)
Comment (j-mai) [reply event/parsing/eventValidator.go:91]: Yeah I think I named it `eventValidator` because there's also the `TimeValidator` struct so I wanted to differentiate the two. But I think I'll change `eventValidator` to `rule` instead.

**Review by kristinaspring — APPROVED** [review]
Looks great! Just one small file name change.
